src/MetadataManager/Metadata.ts

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as flatbuffers from 'flatbuffers';
+import * as fs from 'fs';
+import * as vscode from 'vscode';
+
+import {ToolchainInfo} from '../Backend/Toolchain';
+import * as Circle from '../CircleEditor/circle_schema_generated';
+import {obtainWorkspaceRoot} from '../Utils/Helpers';
+
+import {isOneExplorerTargetFile} from '../Utils/Helpers';
+
+type BuildInfoKeys = 'onecc'|'toolchain'|'cfg';
+
+interface BuildInfoObj {
+  'onecc': string, 'toolchain': ToolchainInfo|undefined, 'cfg': any
+}
+
+export class BuildInfo {
+  private static _map = new Map<string, BuildInfoObj>();
+
+  public static set(path: string, key: BuildInfoKeys, value: any) {
+    const relPath = vscode.workspace.asRelativePath(path);
+    let info = BuildInfo._map.get(relPath);
+    if (info === undefined) {
+      info = {onecc: '', toolchain: undefined, cfg: undefined};
+      BuildInfo._map.set(relPath, info);
+    }
+    info[key] = value;
+  }
+
+  public static save(metaEntry: any, uri: vscode.Uri) {
+    const path = vscode.workspace.asRelativePath(uri);
+    const info = BuildInfo._map.get(path);
+    if (info) {
+      metaEntry['onecc-version'] = info['onecc'];
+      metaEntry['toolchain-version'] = info['toolchain'] ?.version ?.str();
+      metaEntry['cfg-settings'] = info['cfg'];
+    }
+    BuildInfo._map.delete(path);
+  }
+}
+
+export class Operator {
+  public static async get(uri: vscode.Uri) {
+    if (!uri.fsPath.endsWith('.circle')) {
+      return [];
+    }
+    const bytes = new Uint8Array(await vscode.workspace.fs.readFile(uri));
+    const buf = new flatbuffers.ByteBuffer(bytes);
+    const model = Circle.Model.getRootAsModel(buf).unpack();
+    const operators = model.operatorCodes.map((operator) => {
+      Circle.BuiltinOperator[operator.deprecatedBuiltinCode];
+    });
+    return operators;
+  }
+
+  public static async save(metaEntry: any, uri: vscode.Uri) {
+    if (!uri.fsPath.endsWith('.circle')) {
+      return;
+    }
+
+    const stat = fs.statSync(uri.fsPath);
+    const operations: any = {};
+    // TODO deal with large files
+    if (stat.size > 10000) {
```

Comment:
AFAIK, `stat.size` is byte, not KB or MB.
If so, this logic is denying circle file over than 10000byte, which is 9.7KB.
I think 9.7KB is not that large... @.@

Reply:
It seems that we took a long time in the circle operator extraction process using flatbuffer

Of course, it is true that it takes a long time to read a model or circle file or its capacity if it is too large, but specifying its size limit seems to require a lot of testing. 

So for now, it would be better to remove the restrictions.

Reply:
IMO, if it slows down the extension, it's critical, as it gathers up the metadata on extension's startup, the user must wait long. Instead, we decided to support on-click showing of the circle-op information. 
I think it's better to set conservative number for now, but if there is any number you recommends, let us know. @llFreetimell

Reply:
I want to let you know that the function is executed not at the startup but when you run cfg.

And, I think there are 2 options.

> So for now, it would be better to remove the restrictions.

1. Remove restrictions

> I think it's better to set conservative number for now

2. Set conservative number for now

Which one should I choose...?

Reply:
> but if there is any number you recommends, could you let us know? @llFreetimell

Unfortunately, I have no idea for it.
Nevertheless, the reason why I wrote about this was... if 9.7KB is large, almost all circle model files cannot be saved because 9.7KB is really small model.
(FYI, `res/samples/circle/while_000.circle` is 1.5KB even it has less than 10 operators)

Since this restriction is only applied for metadata manger, I am not against with going as it is :)
Please let me know @dayo09's decision, I will follow you!

Reply:
> I want to let you know that the function is executed not at the startup but when you run cfg.

Oh.. Sure. I remember the line. Then, how long does it increase the onecc-running time..? (like, doubled? 20%?)
 
When I skim through the internal model list, 10MB(to support over 60%) or 1MB(more conservatively) looks fine for the restriction.

Plus, please defined the value as a const variable somewhere and add a comment there.

Reply:
The number 10000 was a meaningless value set without testing.
In fact, when I ran a function on a 92MB file, it took me about 30 seconds...
I'm thinking of removing the restriction for consistency with ONEIT team, so if you allow me, I'll fix it and push it.
@dayo09 @llFreetimell

Reply:
> I'm thinking of removing the restriction for consistency with ONEIT team

Since two teams are handling different features, you don't need to think about it :)
I left a comment with maintainer's point of view, not OneIT team!

Reply:
> In fact, when I ran a function on a 92MB file, it took me about 30 seconds...

If so, let's take 1MB restriction. I think affecting the running time is not good. If it's calculated when we click 'View Metadata', it doesn't matter how long time it would take. (<-And you can write down this as TODO.)